### DISCUSSION TITLE
Is there any way to have a interval to set explict_vertical_lines

### DISCUSSION THREAD

**Discussion by youpengbo2018** [jsvine/pdfplumber#495] [Ask for help with specific PDFs]
This is the code that I have used to convert the pdf which doesn't have vertical edge. It works, but I want to loop a file, which has many similar pdf to convert.They have different width. Because of this, I have to change the explict_vertical_lines from 0.9 to 0.88 some time. I want to ask is there any idea that let the pdfplumber to have a interval about the explict vertical lines. So I can loop all the files and get the right content. 

with pdfplumber.open(file_path) as pdf:
    for i in range(pdf.pages[-1].page_number):
        page = pdf.pages[i] 
        for table in page.extract_tables(table_settings={"vertical_strategy": "lines"
            ,"explicit_vertical_lines":[Decimal(page.width)*Decimal(0.07),Decimal(page.width)*Decimal(0.86),Decimal(page.width)*Decimal(0.9)]
            ,"horizontal_strategy": "lines",
        }):

**Comment by samkit-jain** [comment]
Hi @youpengbo2018 Appreciate your interest in the library. Could you please provide more information? The PDF that is causing the issue and the pages where the variation is. Is it only the last line that is changing positions? If so, you can keep the last multiplier of the `explicit_vertical_lines` to the max value you have seen and use a high value for `intersection_x_tolerance` which will ensure that even if there is a significant gap between the last line and the horizontal line, the cell still forms.

**Comment by youpengbo2018** [comment]
[市月度数据(202105).pdf](https://github.com/jsvine/pdfplumber/files/7028861/202105.pdf)
[市月度数据(202106).pdf](https://github.com/jsvine/pdfplumber/files/7028862/202106.pdf)
Thank you for your help.These are some of my Samples. I can extract 202105 file by using the code.
![de3ebd1195729cc12c35e600a9b71a6](https://user-images.githubusercontent.com/61830768/130384094-49c1d629-b6da-44cd-b96f-ff62854c8b4a.png)
When I tried using the same code to extract 202106,it can not give me the entire data.
![cdc4d230aa746bceb24b8bb910d8142](https://user-images.githubusercontent.com/61830768/130384228-35ff33ff-6b68-4bcd-9c7c-f66fc66dbdb4.png)
I want to config a setting which can fit these two PDF. they have different right explict vertical_lines.Could you give me some hint about this.

**Comment by samkit-jain** [comment]
Thanks for sharing the PDF @youpengbo2018 Using the following table settings seems to work on both the PDFs.
```python
{
    "vertical_strategy": "lines",
    "horizontal_strategy": "lines",
    "explicit_vertical_lines": [Decimal(page.width) * Decimal('0.04'), Decimal(page.width) * Decimal('0.94')],
    "intersection_x_tolerance": 30,
}
```

**Reply by youpengbo2018** [reply]
Thank you for the help. It works. And could you describe a little bit about how can you get the intersection_x_tolerance is 30?

**Reply by samkit-jain** [reply]
Hit and trial. I started with 10 and went my way up to 30 in intervals of 10.

**Reply by samkit-jain** [reply]
But a proper approach would be to get the X coordinate of the end of any of the horizontal line segments and the last vertical line segment and take a difference of the two.
